### ISSUE TITLE
Admin Portal

### ISSUE THREAD

**Issue by MaKleSoft** [padloc/padloc#402]
- [ ] GUI for managing server config
- [ ] User management and provisioning
- [ ] Usage stats
- [ ] Advanced Integrations (e.g. various identity providers)
- [ ] View audit logs

**Comment by scottmando2000** [comment]
Will this feature be exclusively to enterprise, or will it also be available to self hosted?
And will there be a way to disable registration with the user management options?

**Comment by MaKleSoft** [comment]
> Will this feature be exclusively to enterprise, or will it also be available to self hosted?

It will be available for self-hosted as well under the same license (AGPL).

> And will there be a way to disable registration with the user management options?

Yes! That falls under "user management and provisioning".

**Comment by MaKleSoft** [comment]
Hey @scottmando2000 - an initial version of the admin portal has been released with v4.2, but it's still very minimal (you can search, browse and delete users and orgs, but thats about it). We'll be adding more features in upcoming releases.

**Comment by MaKleSoft** [comment]
Actually, you can do this now by setting:

```sh
PL_PROVISIONING_BASIC_DEFAULT_STATUS=unprovisioned
```

You can then enable individual accounts by adding an [`AccountProvisioning`](https://github.com/padloc/padloc/blob/b21810cfdaa107638f4c00d99cf59ad92b6c203f/packages/core/src/provisioning.ts#L136) database entry with status `active`.

I know it's not very straightforward, but it's technically possible at least 😄. The admin portal will make this easier by providing a GUI to add/edit these provisioning entries but I can't give you an eta for that yet.

**Comment by ike5** [comment]
@MaKleSoft How would I go about adding an `AccountProvisioning` database entry with the status as `active`? I'm using the docker container version of Padloc on my server and while I've been able to login to the Postgres database to view the tables, I'm not seeing any data itself when trying to select. 

```
postgres=# \l+
                                                                List of databases
   Name    | Owner  | Encoding |  Collate   |   Ctype    | Access privileges |  Size   | Tablespace |                Description                 
-----------+--------+----------+------------+------------+-------------------+---------+------------+--------------------------------------------
 padloc    | padloc | UTF8     | en_US.utf8 | en_US.utf8 |                   | 9517 kB | pg_default | 
 postgres  | padloc | UTF8     | en_US.utf8 | en_US.utf8 |                   | 7901 kB | pg_default | default administrative connection database
 template0 | padloc | UTF8     | en_US.utf8 | en_US.utf8 | =c/padloc        +| 7753 kB | pg_default | unmodifiable empty database
           |        |          |            |            | padloc=CTc/padloc |         |            | 
 template1 | padloc | UTF8     | en_US.utf8 | en_US.utf8 | =c/padloc        +| 7753 kB | pg_default | default template for new databases
           |        |          |            |            | padloc=CTc/padloc |         |            | 
(4 rows)



postgres=# \connect padloc
You are now connected to database "padloc" as user "padloc".
padloc=# \dt
               List of relations
 Schema |        Name         | Type  | Owner  
--------+---------------------+-------+--------
 public | account             | table | padloc
 public | accountprovisioning | table | padloc
 public | auth                | table | padloc
 public | authrequest         | table | padloc
 public | org                 | table | padloc
 public | orgprovisioning     | table | padloc
 public | session             | table | padloc
 public | vault               | table | padloc
(8 rows)

padloc=# select * from account
padloc-# select * from accountprovisioning 
padloc-# 

```

**Comment by BrunoBernardino** [comment]
@ike5 assuming you haven't created an account yet, restart the server with an env var [`PL_PROVISIONING_BASIC_DEFAULT_STATUS=active`](https://github.com/padloc/padloc/blob/main/docs/examples/config/example.env#L283`) which will do that automatically for you.

If you have, it's easier to delete the db and get started again.

To create an account, simply try to login, and you'll be guided through the process.

**Comment by ike5** [comment]
@BrunoBernardino I went ahead and started the project from scratch on a clean server, but I'm encountering an error after inviting users. I'm wondering what I did wrong here.

Steps I took:

1. Configured the following files: `.env`, `nginx.conf`, `docker-compose.yml`, `get-cert.sh`
2. Set `PL_PROVISIONING_BASIC_DEFAULT_STATUS` to `active` in the `.env` file
3. Created the certificate
4. `docker-compose up -d`
5. Login as `PL_ADMIN_EMAIL` user
6. Set `PL_PROVISIONING_BASIC_DEFAULT_STATUS` to `unprovisioned`
7. `docker-compose down`
8. Remove all images
9. `docker-compose up -d`
10. Login as `PL_ADMIN_EMAIL` user

It works at this point for the admin user. I can also see that access is denied to random users who want to create an account on my server. 

Now, after inviting a test user and having them create a password, that user receives an error message (see screenshot). However, I can see in the database that the test user was created successfully and that `"status":"unprovisioned"`. How do I get the invitees to be `active` instead of `unprovisioned` if they are being invited? 

<img width="588" alt="Screenshot 2023-01-09 at 10 31 36 AM" src="https://user-images.githubusercontent.com/31159328/211383671-c2cad0e7-5cf3-49ab-8426-9800bec7723f.png">

**Comment by BrunoBernardino** [comment]
I think that should've been enough. Ideas @MaKleSoft ?

**Comment by MaKleSoft** [comment]
> Now, after inviting a test user and having them create a password, that user receives an error message (see screenshot). However, I can see in the database that the test user was created successfully and that "status":"unprovisioned". How do I get the invitees to be active instead of unprovisioned if they are being invited?

You're almost there! Just manually change `"status": "unprovisioned"` to `"status": "active"` in the database and you should be good to go!

**Comment by ike5** [comment]
@MaKleSoft You mean make a query to change the status to `active` for any new invited user? I'm not sure I can do that, since I'll be leaving my team in the next couple of months, and once I'm gone having them go in to manually query the database and set users to `active` wouldn't be feasible.

**Comment by MaKleSoft** [comment]
This was actually shipped with v4.3.0, although it's currently missing documentation. Some features weren't included, like editing the config via the admin portal, but that will be tracked in separate tickets.
